src/redux/actions.ts

```diff
@@ -0,0 +1,183 @@
+import { AppStore } from 'redux/store';
+import { BASE_URL, DROPDOWN_OPTIONS_DEFINITION } from 'config';
+import {
+  GetMoviesSuccessAction,
+  GET_MOVIES,
+  GET_MOVIES_FAIL,
+  GET_MOVIES_SUCCESS,
+  GetMoviesAction,
+  SetCurrentTabAction,
+  SET_ACTIVE_TAB,
+  GetMovieDetailsAction,
+  GET_MOVIE_DETAILS,
+  GET_MOVIE_DETAILS_SUCCESS,
+  GET_MOVIE_DETAILS_FAIL,
+  GetMovieDetailsSuccessAction,
+  GetMovieDetailsFailAction,
+  GetMoviesFailAction,
+  SET_ACTIVE_SORT_BY,
+  SetCurrentSortByAction,
+  CreateMovieAction,
+  CREATE_MOVIE_SUCCESS,
+  CREATE_MOVIE_FAIL,
+  CreateMovieSuccessAction,
+  CreateMovieFailAction,
+  CREATE_MOVIE,
+  RemoveMovieAction,
+  EditMovieAction,
+  EditMovieFailAction,
+  EditMovieSuccessAction,
+  EDIT_MOVIE_SUCCESS,
+  EDIT_MOVIE_FAIL,
+  EDIT_MOVIE,
+  RemoveMovieSuccessAction,
+  REMOVE_MOVIE_FAIL,
+  RemoveMovieFailAction,
+  REMOVE_MOVIE,
+  REMOVE_MOVIE_SUCCESS,
+} from 'redux/types';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Dispatch = any; // ToDo need to find the proper type
+
+export const setActiveTabAction =
+  (tab: SetCurrentTabAction['payload']) => (dispatch: Dispatch, getStore: () => AppStore) => {
+    const store = getStore();
+
+    dispatch({
+      type: SET_ACTIVE_TAB,
+      payload: tab,
+    });
+
+    dispatch(
+      getMoviesAction({
+        searchBy: 'genres',
+        search: tab === 'all' ? '' : tab,
+        sortBy: store.sortBy,
+      })
+    );
+  };
+
+export const setActiveSortByAction =
+  (sortBy: SetCurrentSortByAction['payload']) => (dispatch: Dispatch, getStore: () => AppStore) => {
+    const store = getStore();
+
+    dispatch({
+      type: SET_ACTIVE_SORT_BY,
+      payload: DROPDOWN_OPTIONS_DEFINITION[sortBy],
+    });
+
+    dispatch(
+      getMoviesAction({
+        searchBy: store.activeTab === 'all' ? '' : 'genres',
+        search: store.activeTab === 'all' ? '' : store.activeTab,
+        sortBy: DROPDOWN_OPTIONS_DEFINITION[sortBy],
+      })
+    );
+  };
+
+export const getMoviesAction =
+  ({ search = '', searchBy = '', sortBy }: GetMoviesAction['payload']) =>
+  (dispatch: Dispatch) => {
+    dispatch({ type: GET_MOVIES });
+
+    fetch(`${BASE_URL}?search=${search}&searchBy=${searchBy}&sortBy=${sortBy}&sortOrder=desc`)
```

Comment:
I would split `fetch` and `dispatch` logic. For data fetching we can create a separate helper